main/src/ca/uwaterloo/flix/language/phase/Finalize.scala

```diff
@@ -469,6 +469,14 @@ object Finalize extends Phase[SimplifiedAst.Root, FinalAst.Root] {
         val t = visitType(tpe)
         FinalAst.Expression.FixpointEntails(e1, e2, t, loc)
 
+      case SimplifiedAst.Expression.FixpointFold(pred, exp1, exp2, exp3, tpe, loc) =>
+        val p = visitPredicateWithParam(pred, m)
+        val e1 = visit(exp1).asInstanceOf[FinalAst.Expression.Var] // TODO not the cleanest
```

Comment:
Change the comment to explain that we have let-bound all the components previously. 
Also add a comment to explain *why*.